pallets/product/src/lib.rs

```diff
@@ -439,6 +445,47 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Update the status of the order
+		///
+		/// * origin: the identity of the stream controller.
+		/// * identifier: unique identifier of the stream.
+		/// * status: stream revocation status (bool).
+		#[pallet::weight(124_410_000 + T::DbWeight::get().reads_writes(1, 2))]
+		pub fn rating(
+			origin: OriginFor<T>,
+			identifier: IdOf<T>,
+			buyer: CordAccountOf<T>,
+			rating: u8,
+		) -> DispatchResult {
+			<T as Config>::EnsureOrigin::ensure_origin(origin)?;
+
+			ensure!(rating < 1 || rating > 10, Error::<T>::InvalidRating);
+
+			let tx_order = <Products<T>>::get(&identifier).ok_or(Error::<T>::OrderNotFound)?;
+			ensure!(tx_order.creator == buyer, Error::<T>::UnauthorizedOperation);
+
+			let block_number = <frame_system::Pallet<T>>::block_number();
+
+			ProductCommit::<T>::store_tx(
+				&tx_order.link.unwrap(),
+				ProductCommit {
+					tx_hash: tx_order.tx_hash.clone(),
+					cid: tx_order.cid.clone(),
+					block: block_number.clone(),
+					commit: ProductCommitOf::Rating,
+				},
+			)?;
+
+			<Products<T>>::insert(
+				&tx_order.link.unwrap(),
+				ProductDetails { block: block_number, rating: Some(rating), ..tx_order },
+			);
+
+			Self::deposit_event(Event::TxReturn(identifier, buyer));
```

Comment:
Just one issue. From subql I would need `ProductRating()`  as the event. That may be better.